### ISSUE TITLE
Cannot access hostmaster

### ISSUE THREAD

**Issue by dserrato** [omega8cc/boa#1629]
After running latest upgrades for both Barracuda and Octopus, all of the sites are active and will come up, but there is no access to the Hostmaster site (i.e. o1.server.mydomain.com)

End up with 502 Bad Gateway error. 

Any ideas?

Thanks so much for any insight someone may have.

**Comment by molaria** [comment]
Same error - restart everything does not solve the problem.

**Comment by dserrato** [comment]
I'm digging and to expand, the 502 Bad Gateway is indicating nginx. 

nginx error log (Adjusted last line that contained clinet info with XXX):

2022/01/17 07:59:46 [emerg] 24562#0: PEM_read_bio_DHparams("/etc/ssl/private/nginx-wild-ssl.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:00:01 [emerg] 865#0: PEM_read_bio_DHparams("/etc/ssl/private/nginx-wild-ssl.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:02:08 [emerg] 8407#0: PEM_read_bio_DHparams("/etc/ssl/private/nginx-wild-ssl.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:03:08 [emerg] 9451#0: PEM_read_bio_DHparams("/etc/ssl/private/nginx-wild-ssl.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:04:01 [emerg] 11716#0: PEM_read_bio_DHparams("/etc/ssl/private/4096.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:04:04 [emerg] 12326#0: PEM_read_bio_DHparams("/etc/ssl/private/4096.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:04:09 [emerg] 12808#0: PEM_read_bio_DHparams("/etc/ssl/private/4096.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:04:09 [emerg] 12879#0: PEM_read_bio_DHparams("/etc/ssl/private/4096.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:04:09 [emerg] 12878#0: PEM_read_bio_DHparams("/etc/ssl/private/4096.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:04:14 [emerg] 21593#0: PEM_read_bio_DHparams("/etc/ssl/private/4096.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:04:15 [emerg] 22434#0: PEM_read_bio_DHparams("/etc/ssl/private/4096.dhp") failed (SSL: error:0909006C:PEM routines:get_name:no start line:Expecting: DH PARAMETERS)
2022/01/17 08:12:42 [crit] 3321#0: *252 SSL_do_handshake() failed (SSL: error:141CF06C:SSL routines:tls_parse_ctos_key_share:bad key share) while SSL handshaking, client: XXX.XXX.XXX.XXX, server: 0.0.0.0:XXX

**Comment by EdNett** [comment]
Hi,

We had this error on a weak test server, and in fact the creation of the 2048 or 4096 openssl  .dhp took a long, long time - it ran for 30 min after the barracuda up-head finished! You might check again in an hour and see if it isn't where nginx expects it. You can also copy an existing one and put it there or use openssl command to place one there.

Check to see if php74-fpm (or php73-fpm ), etc. is stopped. If so, that will show nginx 502 error. (service php74-fpm status)
If nginx won't start it also could be because  php74-fpm.ini file has too high a settign at the very end  (/opt/php74/etc/php74.ini)
apc.shm_size=1024M  - change that 1024M to 64M and restarting php74-fpm should work (BOA will already have restarted it for you the minute you save the php74.ini file. and nginx will probably auto-restart, too.

Good luck

**Comment by dserrato** [comment]
Thanks. I will check that on the next attempt to update BOA. It had been some time and I could not figure it out, so I had to revert to previous snapshot of the server. 

Just to be sure, can you expand on the openssl command that you are referencing. I am not familiar with having to do that, so I could use a reference to using it. And, if I need to use it, will is just override the files with new ssl, or do you need to remove old files and run the command. 

One thing that is different on this server than our other is that there are extra files for the actual server name in addition to the 4096 and nginx-wild-ssl.dhp files. 

in other words, the folder also contains:
o1.myserver.com.crt
o1.myserver.com.dhp
o1.myserver.com.key

Our other server, which also hasn't been updated yet, does not have these extra key and dhp files. Wonder if they are an issue at all or is that how the new updates work.

**Comment by EdNett** [comment]
Just login as root and do:

openssl dhparam -out /etc/ssl/private/4096.dhp 4096

then wait - perhaps as long as an hour - then test to see if nginx and php74-fpm are running:

service nginx status
service php74-fpm status

if they aren't replace "status" with "restart"  and restart them.

**Comment by dserrato** [comment]
@EdNett thank you so much for the information. I've been able to learn from some of your recent posts on these upgrades and it has helped my learning tremendously. 
But, I'm still not getting anywhere. I did use the openssl command and it ran rather quickly and created an output in a few minutes. 

php74-fpm status shows it is running
nginx status give this message: [warn] Nginx Server... found running with processes: 21742 21741 21740 8248 ... (warning)

As well, when I check the nginx error log, there are several connect failures that indicate:
[error] 21740#0: *198 connect() failed (111: Connection refused) while connecting to upstream, client: xx.x.x.xxx ( I removed the numbers, of course)

Upgrades ran great after adding 7.4 for php version in the baracuda and the octopus.cnf, as well as the cli.info and fpm.info files in the o1 control folder. All of the sites are running great, but still no access to the main hostmaster site.

**Comment by omega8cc** [comment]
You need to check PHP errors in the `/var/log/php/` to find the culprit.

**Comment by omega8cc** [comment]
It's for example possible that you have specified PHP version for Octopus which is not installed or broken for some reason. Just check the error logs.

**Comment by dserrato** [comment]
Here is how the upgrade finally succeeded. 

I normally use silent mode, but this time did not. Not saying that changed anything, just pointing out the steps in the process this time. 

Changed .barracuda.cnf to the following for php version:

_PHP_CLI_VERSION=7.4
_PHP_FPM_VERSION=7.4
_PHP_MULTI_INSTALL=7.4
_PHP_SINGLE_INSTALL=7.4
_PHP_MULTI_INSTALL=7.4

Changed o1.octopus.cnf php versions to
_PHP_FPM_VERSION=7.4
_PHP_CLI_VERSION=7.4

In the o1/static/control/ folder, changed the cli and fpm files to 7.4

Ran the first two steps:
```
cd;wget -q -U iCab http://files.aegir.cc/BOA.sh.txt;bash BOA.sh.txt
barracuda up-head
```

Noted that the upgrade process went through the php 7.3 and php 7.2 install/upgrade

Barracuda upgrade finished successfully, but ALL sites on the server were down. 

Server was up and running fine 
php74-fpm was running
php73 and php72 were not running
nginx status was running
But all sites had Bad Gateway nginx errors

Checked the php error log and found:
```
[20-Jan-2022 05:42:02 UTC] PHP Warning:  PHP Startup: uploadprogress: Unable to initialize module
Module compiled with module API=20180731
PHP    compiled with module API=20190902
These options need to match
 in Unknown on line 0
[20-Jan-2022 05:42:02 UTC] PHP Warning:  PHP Startup: imagick: Unable to initialize module
Module compiled with module API=20180731
PHP    compiled with module API=20190902
These options need to match
 in Unknown on line 0
[20-Jan-2022 05:42:02 UTC] PHP Warning:  PHP Startup: jsmin: Unable to initialize module
Module compiled with module API=20180731
PHP    compiled with module API=20190902
These options need to match
 in Unknown on line 0
[20-Jan-2022 05:42:02 UTC] PHP Warning:  PHP Startup: redis: Unable to initialize module
Module compiled with module API=20180731
PHP    compiled with module API=20190902
These options need to match
 in Unknown on line 0
[20-Jan-2022 05:42:02 UTC] PHP Warning:  PHP Startup: yaml: Unable to initialize module
Module compiled with module API=20180731
PHP    compiled with module API=20190902
These options need to match
 in Unknown on line 0
```


Wanted to forge on and see if the rest of the upgrade process would help, so...

Ran `barracuda up-head system` - completed successfully


Went to run octopus upgrade and it failed because of incorrect versions

Opened o1.octopus.cnf and the php versions were back to 5.2

Changed them to 7.4

Failed again, so I went to the o1/static/control folder and had to change the cli and fpm files to 7.4 again. 

Ran `octopus up-head all force`

Finished successfully and all of the sites started coming up

Ran the two last commands:
`bash /var/xdrago/manage_ltd_users.sh`
`bash /var/xdrago/daily.sh `

Successful upgrade and so far everything is running great. It's almost midnight, so I need to get some sleep. 
Thank you @EdNett  and @omega8cc team for all of the feedback and assistance. 
-Dan

**Comment by dserrato** [comment]
Just a quick update for possible feedback. The hostmaster site and the websites are running great and VERY quick. 

But, I cannot complete a clone or migrate. Wondering if should run another round of upgrades to barracuda and octopus to make sure everything is set right. Would that make sense?

**Comment by omega8cc** [comment]
@dserrato What you mean by "cannot complete a clone or migrate"? Please provide more details, errors reported in te failed tasks etc.

**Comment by dserrato** [comment]
This morning I ran through the upgrade process again and there are now successful clones and migrates on some test sites, which was so great. But, did get a situation with a migrate stalling and referencing MyQuick, so I actually just posted an issue separately in an effort to try and not muck up this issue too much. I hope that's ok.
